services/dynamic-sidecar/tests/unit/test_core_docker_utils.py

```diff
@@ -46,9 +46,9 @@ async def test_volume_with_label(
 
 
 async def test_volume_label_missing(run_id: UUID) -> None:
-    with pytest.raises(VolumeNotFoundError) as info:
+    with pytest.raises(VolumeNotFoundError) as exc_info:
         await get_volume_by_label("not_exist", run_id)
-    assert info.value.args[0] == (
-        f"Expected 1 volume with source_label='not_exist', "
-        f"run_id=UUID('{run_id}'), found: volumes=[]"
-    )
+
+    error_msg = f"{exc_info.value}"
+    assert f"{run_id}" in error_msg
+    assert "not_exist" in error_msg
```

Comment:
Why not assert against the full message? I though we said this would be the better alternative?

Reply:
IMO when an error message **is not structured**  asserting exact messages has low RoI since we tend to rephrase them. Checking keywords in them and has the same test effect